dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/preheat/DefaultPreheatService.java

```diff
@@ -709,53 +714,42 @@ else if ( Map.class.isInstance( object ) )
             map.put( objectClass, refMap );
 
-            for ( IdentifiableObject object : identifiableObjects )
-            {
-                refMap.put( object.getUid(), new HashMap<>() );
-
-                properties.forEach( p ->
-                {
-                    if ( !p.isCollection() )
-                    {
-                        IdentifiableObject reference = ReflectionUtils.invokeMethod( object, p.getGetterMethod() );
+            for ( IdentifiableObject object : identifiableObjects ) {
+                if (object != null) {
+                    refMap.put(object.getUid(), new HashMap<>());
 
-                        if ( reference != null )
-                        {
-                            try
-                            {
-                                IdentifiableObject identifiableObject = (IdentifiableObject) p.getKlass().newInstance();
-                                mergeService.merge( new MergeParams<>( reference, identifiableObject ) );
-                                refMap.get( object.getUid() ).put( p.getName(), identifiableObject );
-                            }
-                            catch ( InstantiationException | IllegalAccessException ignored )
-                            {
-                            }
-                        }
-                    }
-                    else
+                    properties.forEach(p ->
                     {
-                        Collection<IdentifiableObject> refObjects = ReflectionUtils.newCollectionInstance( p.getKlass() );
-                        Collection<IdentifiableObject> references = ReflectionUtils.invokeMethod( object, p.getGetterMethod() );
-
-                        if ( references != null )
-                        {
-                            for ( IdentifiableObject reference : references )
-                            {
-                                try
-                                {
-                                    IdentifiableObject identifiableObject = (IdentifiableObject) p.getItemKlass().newInstance();
-                                    mergeService.merge( new MergeParams<>( reference, identifiableObject ) );
-                                    refObjects.add( identifiableObject );
+                        if (!p.isCollection()) {
+                            IdentifiableObject reference = ReflectionUtils.invokeMethod(object, p.getGetterMethod());
+
+                            if (reference != null) {
+                                try {
+                                    IdentifiableObject identifiableObject = (IdentifiableObject) p.getKlass().newInstance();
+                                    mergeService.merge(new MergeParams<>(reference, identifiableObject));
+                                    refMap.get(object.getUid()).put(p.getName(), identifiableObject);
+                                } catch (InstantiationException | IllegalAccessException ignored) {
                                 }
```

Comment:
Just wondering, why do we need to change the preheat service?

Reply:
Fixed. Chnages reverted